### ISSUE TITLE
Allow value of $let to be evaluated

### ISSUE THREAD

**Issue by djmitche** [taskcluster/json-e#249]
```yaml
$let:
  $if: something
  then: {a: 10, b: 20}
  else: {a: 20, b: 10}
in:
  $eval: 'a + b'
```

As of right now, this is not legal, because the value of the `$let` is not evaluated -- it must be an actual object, the *values* of which are evaluated.  But we could just evaluate the whole thing and require that the result be an object (which is not then re-evaluated)

**Comment by terrameijar** [comment]
@owlishDeveloper I would like to try my hand at this issue. Could you please give me pointers on how to approach it? What is the best way to make sure `$let` gets evaluated?

**Comment by stuartelimu** [comment]
can i work on this issue

**Comment by owlishDeveloper** [comment]
@terrameijar before starting working on the issues, you have to setup your development environment, successfully run the tests and send me the screenshots of the successful test runs. Thank you!

@stuartelimu issues can be worked on only one at a time. You already have one assigned. Thank you!

**Comment by owlishDeveloper** [comment]
@richardgreg did you mean you would like to work on this issue?

**Comment by vishakhanihore** [comment]
@owlishDeveloper I have successfully set up the project. Can you please tell me your email-id ?

**Comment by osumgbachiamaka** [comment]
@owlishDeveloper can i have this task please

**Comment by vishakhanihore** [comment]
@owlishDeveloper Sent you ss, Please assign

**Comment by richardgreg** [comment]
@owlishDeveloper yes, please!

**Comment by owlishDeveloper** [comment]
@richardgreg asked first (terrameijar already has another issue assigned), so this issue goes to richardgreg!

**Comment by richardgreg** [comment]
@djmitche pls could you provide the result for the JSON-e statement you wrote, assuming the issue has been resolved?

**Comment by owlishDeveloper** [comment]
@richardgreg  in that particular example the result should be `30`. Let me amend the example a bit:

```yml
$let:
  $if: something
  then: {a: 10, b: 20}
  else: {a: 10, b: 10}
in:
  $eval: 'a + b'
```
Here, is `something` evaluates to `true`, the result should be `30`, otherwise `20`

**Comment by gregsdennis** [comment]
What's the difference between this and #212?  It seems to me that implementing this issue overturns that decision.

**Comment by djmitche** [comment]
That's a good point - I don't think we had considered that this decision had been made in the opposite direction already.

That said, the argument in #212 is basically "nobody should want this" and "we may want to do static analysis", but the example in the first comment is actually pretty reasonable: often all of the variables depend on the same condition. And, we aren't doing static analysis. So I'd say it was OK to override that decision at the time.
